### ISSUE TITLE
ConnectionExt cannot be imported automatically

### ISSUE THREAD

**Issue by mahkoh** [psychon/x11rb#661]
There are multiple traits called ConnectionExt, multiple of which often must be used within the same module. IDEs are unable to import these traits automatically due to the name conflicts or will import them but cause errors at compile time. This would not be a problem if these traits were prefixed with their protocol.

**Comment by psychon** [comment]
> IDEs are unable to import these traits automatically due to the name conflicts or will import them but cause errors at compile time. 

Just so that I understand this correctly: The problem is not that the IDE does not find the right trait, the problem is that it could end up with something like the following (which fails to compile because there are two `ConnectionExt`s in scope), right?
```rust
use x11rb::protocol::{shm::ConnectionExt, sync::ConnectionExt};
```

**Comment by mahkoh** [comment]
That's correct. CLion is also able to find all of the methods. The problem happens once I try to compile the program.

**Comment by eduardosm** [comment]
You probably want to do something like this (note the `as _`):
```rust
use x11rb::protocol::{shm::ConnectionExt as _, sync::ConnectionExt as _};
```
That will bring trait methods into scope, but not the trait name itself.
